tensorflow_datasets/image/div2k.py

```diff
@@ -0,0 +1,150 @@
+
+_CITATION = """@InProceedings{Ignatov_2018_ECCV_Workshops,
+author = {Ignatov, Andrey and Timofte, Radu and others},
+title = {PIRM challenge on perceptual image enhancement on smartphones: report},
+booktitle = {European Conference on Computer Vision (ECCV) Workshops},
+url = "http://www.vision.ee.ethz.ch/~timofter/publications/Agustsson-CVPRW-2017.pdf",
+month = {January},
+year = {2019}
+}
+"""
+
+_DESCRIPTION = """
+DIV2K dataset: DIVerse 2K resolution high quality images as used for the challenges @ NTIRE (CVPR 2017 and CVPR 2018) and @ PIRM (ECCV 2018)
+"""
+
+_DL_URL = "https://data.vision.ee.ethz.ch/cvl/DIV2K/"
+
+_DL_URLS = {
+    "train_hr": _DL_URL + "DIV2K_train_HR.zip",
+    "valid_hr": _DL_URL + "DIV2K_valid_HR.zip",
+    "train_bicubic_x2": _DL_URL + "DIV2K_train_LR_bicubic_X2.zip",
+    "train_unknown_x2": _DL_URL + "DIV2K_train_LR_unknown_X2.zip",
+    "valid_bicubic_x2": _DL_URL + "DIV2K_valid_LR_bicubic_X2.zip",
+    "valid_unknown_x2": _DL_URL + "DIV2K_valid_LR_unknown_X2.zip",
+    "train_bicubic_x3": _DL_URL + "DIV2K_train_LR_bicubic_X3.zip",
+    "train_unknown_x3": _DL_URL + "DIV2K_train_LR_unknown_X3.zip",
+    "valid_bicubic_x3": _DL_URL + "DIV2K_valid_LR_bicubic_X3.zip",
+    "valid_unknown_x3": _DL_URL + "DIV2K_valid_LR_unknown_X3.zip",
+    "train_bicubic_x4": _DL_URL + "DIV2K_train_LR_bicubic_X4.zip",
+    "train_unknown_x4": _DL_URL + "DIV2K_train_LR_unknown_X4.zip",
+    "valid_bicubic_x4": _DL_URL + "DIV2K_valid_LR_bicubic_X4.zip",
+    "valid_unknown_x4": _DL_URL + "DIV2K_valid_LR_unknown_X4.zip",
+    "train_bicubic_x8": _DL_URL + "DIV2K_train_LR_x8.zip",
+    "valid_bicubic_x8": _DL_URL + "DIV2K_valid_LR_x8.zip",
+    "train_realistic_mild_x4": _DL_URL + "DIV2K_train_LR_mild.zip",
+    "valid_realistic_mild_x4": _DL_URL + "DIV2K_valid_LR_mild.zip",
+    "train_realistic_difficult_x4": _DL_URL + "DIV2K_train_LR_difficult.zip",
+    "valid_realistic_difficult_x4": _DL_URL + "DIV2K_valid_LR_difficult.zip",
+    "train_realistic_wild_x4": _DL_URL + "DIV2K_train_LR_wild.zip",
+    "valid_realistic_wild_x4": _DL_URL + "DIV2K_valid_LR_wild.zip",
+}
+
+_DATA_OPTIONS = ["bicubic_x2", "bicubic_x3", "bicubic_x4", "bicubic_x8",
+                 "unknown_x2", "unknown_x3", "unknown_x4",
+                 "realistic_mild_x4", "realistic_difficult_x4",
+                 "realistic_wild_x4"]
+
+class Div2kConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Div2k."""
+
+  def __init__(self, data, **kwargs):
+    """Constructs a Div2kConfig."""
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+    name = kwargs.get("name")
+    if name is None:
+      name = data
+    kwargs["name"] = name
+
+    description = kwargs.get("description")
+    if description is None:
+      description = "Uses %s data." % data
+    kwargs["description"] = description
+
+    super(Div2kConfig, self).__init__(**kwargs)
+    self.data = data
+
+  def download_urls(self):
```

Comment:
you can define it as a property, so that while getting the download link on line 118, you can just refer it as
`self.builder_config.download_urls` instead using a function call